backend/controllers/goodVideoController.ts

```diff
@@ -2,38 +2,45 @@ import express from 'express'
 import { GoodVideoApplicationService } from '../application/goodVideo'
 
 export class GoodVideoController {
-  async getLikeCount(req: express.Request, res: express.Response) {
+  private goodVideoApplicationService: GoodVideoApplicationService
+
+  constructor() {
+    this.goodVideoApplicationService = new GoodVideoApplicationService()
+    this.goodVideo = this.goodVideo.bind(this)
+    this.cancelGoodVideo = this.cancelGoodVideo.bind(this)
+    this.getGoodCount = this.getGoodCount.bind(this)
+  }
+
+  async getGoodCount(req: express.Request, res: express.Response) {
     try {
       const { videoId } = req.params
-      const likeCount = await new GoodVideoApplicationService().getLikeCount(
+      const goodCount = await this.goodVideoApplicationService.getGoodCount(
         videoId,
       )
-      res.status(200).json(likeCount)
+      res.status(200).json(goodCount)
     } catch (error) {
-      res.status(500)
-      throw error
+      res.status(500).json({ message: 'エラーが発生しました' })
     }
   }
-
   async goodVideo(req: express.Request, res: express.Response) {
     try {
       const { userId, videoId } = req.body
-      await new GoodVideoApplicationService().goodVideo(userId, videoId)
-      res.status(200).json({ message: 'Good video' })
+
+      await this.goodVideoApplicationService.goodVideo(userId, videoId)
+      res.status(200).json({ message: 'いいねをしました' })
     } catch (error) {
-      res.status(500)
-      throw error
+      res.status(500).json({ message: 'いいねした際にエラーが発生しました' })
```

Comment:
コンソールエラー部分はここのエラーです。

Reply:
ここのエラーにはどんなエラーが入ってきたのでしょうか？
```suggestion
console.log('error',error);

      res.status(500).json({ message: 'いいねした際にエラーが発生しました' })
```

Reply:
こちらになります。
error PrismaClientKnownRequestError: 
api  | Invalid `prisma.goodVideo.create()` invocation in
api  | /app/application/goodVideo.ts:50:46
api  | 
api  |   47   })
api  |   48   return goodVideo
api  |   49 }
api  | → 50 const goodVideo = await prisma.goodVideo.create(
api  | Unique constraint failed on the constraint: `GoodVideo_user_id_video_id_key`
api  |     at si.handleRequestError (/app/node_modules/@prisma/client/runtime/library.js:125:6817)
api  |     at si.handleAndLogRequestError (/app/node_modules/@prisma/client/runtime/library.js:125:6151)
api  |     at si.request (/app/node_modules/@prisma/client/runtime/library.js:125:5859)
api  |     at async l (/app/node_modules/@prisma/client/runtime/library.js:130:9805) {
api  |   code: 'P2002',
api  |   clientVersion: '5.8.0',
api  |   meta: { modelName: 'GoodVideo', target: 'GoodVideo_user_id_video_id_key' }
api  | }

調べたところ、Prismaが提供するエラーで、一般的にはデータベースの一意性制約に違反した場合に表示されます。エラーメッセージの内容を見ると、具体的にはGoodVideoモデルのuser_idとvideo_idのユニーク制約が失敗したことを示しています。

エラーメッセージの詳細:

code: 'P2002': Prismaのエラーコードで、一意性制約違反を示します。
clientVersion: '5.8.0': 使用している Prisma クライアントのバージョン。
meta: { modelName: 'GoodVideo', target: 'GoodVideo_user_id_video_id_key' }: エラーが発生したモデルと具体的な制約。
エラーが発生しているコード:
const goodVideo = await prisma.goodVideo.create({
  data: {
    user_id: userId,
    video_id: videoId,
  },
});
これにより、GoodVideoモデルにおいて、user_idとvideo_idの組み合わせが既に存在しているため、一意性制約違反が発生しています。

つまりすでにDBにIDがあるためダメですってことですか？